### PR TITLE
Random personnel background generator

### PR DESCRIPTION
**Pull request by sixlettervariables**
This abstracts personnel generation to support adding random factions/planets.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/SpecialAbility.java [hunk 1]
```diff
@@ -161,14 +161,39 @@ public boolean isEligible(Person p) {
...
-    
+
+    public boolean isEligible(boolean isClanner, Skills skills, PersonnelOptions options) {
+        for(SkillPrereq sp : prereqSkills) {
+            if(!sp.qualifies(skills)) {
+                return false;
+            }
+        }
+        for(String ability : prereqAbilities) {
+            //TODO: will this work for choice options like weapon specialist?
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/SpecialAbility.java]: It'll probably detect that the person has the category of SPA, but not the specific sub-type.

Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/personnel/SpecialAbility.java]: This is all existing code and not my TODO's, but good to know it won't work for a sub type.

**Comment by sixlettervariables** [comment]
This should be ready for play testing its reasonableness. I need to add its campaign options to customize and more Java Docs.

File: MekHQ/src/mekhq/campaign/universe/RangedFactionSelector.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+ * An implementation of {@link AbstractFactionSelector} which chooses
+ * a faction from a defined range of planets.
+ */
+public class RangedFactionSelector extends AbstractFactionSelector {
+
+    private final int range;
+
+    private DateTime cachedDate;
+    private PlanetarySystem cachedSystem;
+    private TreeMap<Double, Faction> cachedFactions;
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/universe/RangedFactionSelector.java]: An explanation of the data structure would be helpful.

**Pull request merged**